http/src/main/java/io/micronaut/http/cachecontrol/CacheControl.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright 2017-2025 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.cachecontrol;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * HTTP Cache-Control header value.
+ */
+public final class CacheControl {
+    private static final String EQUAL = "=";
+    private final List<CacheControlResponseDirective> responseDirectives;
+
+    /**
+     *
+     * @param responseDirectives response Directives
+     */
+    CacheControl(List<CacheControlResponseDirective> responseDirectives) {
```

Comment:
seems like this can be a `record`

Reply:
I changed to use a record https://github.com/micronaut-projects/micronaut-core/pull/11701/commits/2cfafa6f4c54bf9d220f529862ff7b0c2934fae2 The drawback is that with a record the construct has public visibility. Same as the record.